components/CreateEditDialog.tsx

```diff
@@ -0,0 +1,109 @@
+import { observer } from "mobx-react-lite";
+import { useContext, useState } from "react";
+import { v4 as newUuidV4 } from "uuid";
+import { Button, Dialog, Input, TextArea, VisuallyHidden } from "tamagui";
+
+import { Todo, TodoContext } from "../stores/Todo.store";
+import { NewTodoButton } from "./NewButton";
+import { runInAction } from "mobx";
+
+type BaseDialogProps = {
+  dialogTitle: string;
+  dialogDescription: string;
+  defaultTitle?: string;
+  defaultText?: string;
+  onPress: (title: string, text: string) => void;
+  trigger: React.FunctionComponent<{ onPress: () => void }>;
+};
+
+const BaseDialog = observer(function BaseDialog({
+  dialogTitle,
+  dialogDescription,
+  defaultTitle,
+  defaultText,
+  trigger: Trigger,
+  onPress,
+}: BaseDialogProps) {
+  const [isOpen, setOpen] = useState(false);
+  const [title, setTitle] = useState(defaultTitle ?? "");
+  const [text, setText] = useState(defaultText ?? "");
+
+  return (
+    <Dialog open={isOpen} defaultOpen={false}>
+      <Dialog.Trigger asChild>
+        <Trigger onPress={() => setOpen(true)} />
+      </Dialog.Trigger>
+      <Dialog.Portal>
+        <Dialog.Content>
+          <Dialog.Title>{dialogTitle}</Dialog.Title>
+          <VisuallyHidden>
+            <Dialog.Description>{dialogDescription}</Dialog.Description>
+          </VisuallyHidden>
+
+          <Input value={title} onChangeText={setTitle} />
+          <TextArea value={text} onChangeText={setText} />
+
+          <Dialog.Close asChild>
+            <Button
+              onPress={() => {
+                onPress(title, text);
+                setOpen(false);
+              }}
+            >
+              Save to-do item.
+            </Button>
+          </Dialog.Close>
+        </Dialog.Content>
+      </Dialog.Portal>
+    </Dialog>
+  );
+});
+
+export const NewDialog = function NewDialog() {
+  const store = useContext(TodoContext)!;
+
+  return (
+    <BaseDialog
+      dialogTitle="Create a new to-do!"
+      dialogDescription="Enter your text to create a new item to-do."
+      trigger={NewTodoButton}
+      onPress={(title, text) => {
```

Comment:
All this logic here can be moved away from JSX and into the store. The store should have an 'addToDo' to that accepts the necessary information and create a new TODO item.

Reply:
Done